Move sizeLimit one level above in upload provider

Marc-Roig opened the pull request:
### What does it do?
Updates in the upload plugin Guide:

- Add migration guide for the next minor release.
- Update of "Max file size" in upload plugin.

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/pull/14398
https://github.com/strapi/strapi/issues/13282

strapi-cla commented:
[![CLA assistant check](https://cla.strapi.io/pull/badge/not_signed)](https://cla.strapi.io/strapi/documentation?pullRequest=1133) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.strapi.io/strapi/documentation?pullRequest=1133) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.strapi.io/check/strapi/documentation?pullRequest=1133) it.</sub>

Marc-Roig commented:
I added a new migration file explaining:
- people should move the sizeLimit configuration (if being used)
- also be careful if the sizeLimit value was in kilobytes.

What do you think?

pwizla commented on docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.3.x-to-4.4.x.md:
```diff
@@ -0,0 +1,108 @@
+---
+title: Migrate from 4.2.x to 4.3.x - Strapi Developer Docs
```
```suggestion
title: Migrate from 4.3.x to 4.4.x - Strapi Developer Docs
```

:-)

pwizla commented on docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.3.x-to-4.4.x.md:
```diff
@@ -0,0 +1,108 @@
+The Strapi v4.3.x to v4.4.x migration guide upgrades versions of v4.3.x and above to v4.4.x. This migration guide is needed for all users who were limiting media size for the local upload provider. The migration to 4.4.x consists of 3 steps:
+
+- Upgrading the application dependencies
+- Updating the local upload provider sizeLimit
```
```suggestion
- Updating the local upload provider `sizeLimit`
```

I'll also mention/suggest it with every occurrence of the word. We should use inline code formatting as it's the actual name of a property (would be "size limit" in 'pure' English, but `sizeLimit` for the property name). 😊

pwizla commented on docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.3.x-to-4.4.x.md:
```diff
@@ -0,0 +1,108 @@
+
+## Updating the sizeLimit provider configuration
+
+This step is only required if you were using the [sizeLimit configuration](https://docs.strapi.io/developer-docs/latest/plugins/upload.html#max-file-size) for your upload provider.
```
```suggestion
This step is only required if you were using the [`sizeLimit` configuration](/developer-docs/latest/plugins/upload.md#max-file-size) for your upload provider.
```

Updated the URL because:
- if you keep the `https://docs.strapi.io` part, it's considered by our tool as an external link and appends a specific icon for external links next to it ^^
- updated the `.html` extension to `.md` so that our broken links checking tool won't complain (you can test it with `yarn check-links` locally)

pwizla commented on docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.3.x-to-4.4.x.md:
```diff
@@ -0,0 +1,108 @@
+This step is only required if you were using the [sizeLimit configuration](https://docs.strapi.io/developer-docs/latest/plugins/upload.html#max-file-size) for your upload provider.
+
+:::caution
+The docs required the sizeLimit to be in bytes, but it was actually in kilobytes. This is now fixed, and the limit will be interpreted as bytes.
```
```suggestion
The documentation required the `sizeLimit` to be in bytes, but it was actually in kilobytes. This is now fixed, and the limit will be interpreted as bytes.
```

Per our [12 rules](https://handbook.strapi.io/user-success-manual/12-rules-of-technical-writing) :-) (see rule #9 — we avoid abbreviations, even the 'obvious' ones: app → application, docs → documentation, config → configuration, etc. 😊 )

pwizla commented on docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.3.x-to-4.4.x.md:
```diff
@@ -0,0 +1,108 @@
+If you, for some reason, were limiting the file size to kilobytes, you should update the value to be in bytes.
+:::
+
+We recommend to move the sizeLimit outside the provider options like the following, as it will be deprecated in the next major version.
```
```suggestion
We recommend to move the `sizeLimit` outside the provider options like the following, as it will be deprecated in the next major version.
```

pwizla commented on docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.3.x-to-4.4.x.md:
```diff
@@ -0,0 +1,108 @@
+
+To change the script:
+
+1. In the `./config/plugins.js` file, Identify the upload configuration if you have one.
```
```suggestion
1. In the `./config/plugins.js` file, identify the upload configuration if you have one.
```

pwizla commented on docs/developer-docs/latest/update-migration-guides/migration-guides/v4/migration-guide-4.3.x-to-4.4.x.md:
```diff
@@ -0,0 +1,108 @@
+To change the script:
+
+1. In the `./config/plugins.js` file, Identify the upload configuration if you have one.
+2. Move your sizeLimit, if you have one, one level above providerOptions.
```
```suggestion
2. (_optional_) If you have a `sizeLimit`, move it one level above `providerOptions`.
```

pwizla reviewed (changes requested):
Thank you very much and congrats for your very first PR to our documentation repo, @Marc-Roig! 🥳 
Good job.
I added some suggestions, those are mostly formatting-related. Happy to help if you have any questions. 🤗

Marc-Roig commented:
> Thank you very much and congrats for your very first PR to our documentation repo, @Marc-Roig! 🥳
Good job.
I added some suggestions, those are mostly formatting-related. Happy to help if you have any questions. 🤗

Thank you @pwizla !! I still have to get used to the convention and writing more docs 🤓 .
I have committed the changes you requested.

Marc-Roig commented:
Also, as this requires a migration guide, I guessed it should be scheduled for 4.4.X , I do not know if that should be the case.

StrapiShaun commented:
> Also, as this requires a migration guide, I guessed it should be scheduled for 4.4.X , I do not know if that should be the case.

The PR in the main repo is flagged for v4.3.9. If it is included in that patch release the migration guide should be similarly named and released.

pwizla commented:
Good point from Shaun. @Marc-Roig will this change be released with 4.3.9 or with 4.4.0?

Marc-Roig commented:
> > Also, as this requires a migration guide, I guessed it should be scheduled for 4.4.X , I do not know if that should be the case.
> 
> The PR in the main repo is flagged for v4.3.9. If it is included in that patch release the migration guide should be similarly named and released.

you are completely right.
This change will be released in 4.4.0, I changed the flag in the main repo PR

pwizla commented:
Thanks for the clear answer, Marc. I've just updated the documentation milestone as well, then.

pwizla reviewed (dismissed):
LGTM, thanks for the updates, and congrats again, great job!
Please let me know when the corresponding `strapi/strapi` PR is merged, and I'll merge this one.

pwizla commented:
Hi @Marc-Roig I updated the docs milestone to 4.4.3 because the strapi/strapi PR does not seem to be merged yet. But if it's deployed sooner let me know and we could have an urgent deployment (product & docs v4.4.2 is deployed in 2 hours). Thanks again!

Marc-Roig commented:
@pwizla Thank you! I am waiting for QA approval before the merge. There is no rush for this !

Marc-Roig commented:
Sorry for the wait , this will now be ready to merge in 4.6.1 . 
I have moved the migration steps to the `migration-guid-4.5.1-to-4.6.1.md` file, introduced in https://github.com/strapi/documentation/pull/1430. 

So we will have to merge that PR first ⚠️

pwizla reviewed (approved):
LGTM now. Thanks again, Marc!
